src/network/networkd-address.c

```diff
@@ -1082,5 +1082,12 @@ int address_section_verify(Address *address) {
                                          address->section->filename, address->section->line);
         }
 
+        if (address->family != AF_INET && address->ipv4_duplicate_address_detection) {
```

Comment:
should there be a similar check for ipv6 DAD on non-ipv6 addresses?

Reply:
NODAD flag for IPv4 address should be silently ignored by kernel. So, I think it is not necessary.